http/parse.go

```diff
@@ -1,42 +1,38 @@
 package http
 
 import (
-	"errors"
+	"context"
 	"fmt"
+	"io/ioutil"
 	"mime"
 	"net/http"
+	"strconv"
 	"strings"
 
 	"github.com/ipfs/go-ipfs-cmdkit"
 	"github.com/ipfs/go-ipfs-cmdkit/files"
 	cmds "github.com/ipfs/go-ipfs-cmds"
-	path "github.com/ipfs/go-ipfs/path"
 )
 
-// Parse parses the data in a http.Request and returns a command Request object
-func Parse(r *http.Request, root *cmds.Command) (cmds.Request, error) {
-	if !strings.HasPrefix(r.URL.Path, ApiPath) {
-		return nil, errors.New("Unexpected path prefix")
+// parseRequest parses the data in a http.Request and returns a command Request object
+func parseRequest(ctx context.Context, r *http.Request, root *cmds.Command) (*cmds.Request, error) {
```

Comment:
Why unexport this?

Reply:
Because it didn't make sense to me why it should be exported (especially since parseResponse was not exported)